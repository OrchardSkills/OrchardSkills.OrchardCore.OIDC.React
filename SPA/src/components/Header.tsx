import * as React from "react";
import { BrowserRouter as Router, Link } from "react-router-dom";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import Container from "react-bootstrap/Container";
export interface IHeaderProps {
  pageTitle: string;
  logoSrc: any;
  authUser: boolean
}

export default function Header(props: IHeaderProps) {
  return (
    <header className="App-header">
      <img src={props.logoSrc} className="App-logo" alt="logo" />
      <h1 className="App-title">{props.pageTitle}</h1>
      <Navbar bg="dark" variant="dark">
        {props.authUser ? (
          <Container>
            <Navbar.Brand>
              <Link to={"/create-subscriber"} className="nav-link">
                Subscribers
            </Link>
            </Navbar.Brand>
            <Nav className="justify-content-end">
              <Nav>
                <Link to={"/create-subscriber"} className="nav-link">
                  Create
              </Link>
                <Nav>
                  <Link to={"/subscriber-list"} className="nav-link">
                    List
                </Link>
                </Nav>
              </Nav>
            </Nav>
          </Container>
        ) : (
            <div className="text-center">
              Please login to enable CRUD operations.
            </div>
          )}

      </Navbar>
    </header>
  );
}
