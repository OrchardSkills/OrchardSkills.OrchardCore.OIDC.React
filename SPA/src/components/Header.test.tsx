import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import logo from '../logo.svg';
import Header from './Header';

describe(Header, () => {
  const pageTitle = 'OpenID Connect (OIDC) javascript client with React';

  const component = shallow(<Header pageTitle={pageTitle} logoSrc={logo} />);

  it('renders and matches our snapshot', () => {
    const headerComponent = renderer.create(<Header pageTitle={pageTitle} logoSrc={logo} />);
    const tree = headerComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the image and h1', () => {
    expect(component.find('img')).toHaveLength(1);
    expect(component.find('h1')).toHaveLength(1);
  });

  it('h1 contains correct title', () => {
    expect(component.find('h1').text()).toEqual(pageTitle);
    expect(component.contains(pageTitle)).toEqual(true);
  });
});
