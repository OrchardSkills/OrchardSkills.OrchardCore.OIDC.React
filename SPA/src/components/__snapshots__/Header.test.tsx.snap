// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders and matches our snapshot 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <h1
    className="App-title"
  >
    OpenID Connect (OIDC) javascript client with React
  </h1>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      className="text-center"
    >
      Please login to enable CRUD operations.
    </div>
  </nav>
</header>
`;
